### ISSUE TITLE
Feature Request: License Manager

### ISSUE THREAD

**Issue by gazoakley** [terraform-providers/terraform-provider-aws#6654]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* `aws_licensemanager_license_configuration` (New resource)
* `aws_licensemanager_settings` (New resource)
* `aws_launch_template` (Affected resource)

### Potential Terraform Configuration

```hcl
resource "aws_licensemanager_license_configuration" "example" {
  name                     = "Example"
  description              = "Example"
  license_count            = 10
  license_count_hard_limit = true
  license_counting_type    = "Socket"

  license_rules = [
    "#minimumSockets=3"
  ]

  tags {
    foo = "barr"
  }
}
```

### References

* https://docs.aws.amazon.com/license-manager/latest/APIReference/API_CreateLicenseConfiguration.html
* https://docs.aws.amazon.com/license-manager/latest/APIReference/API_GetLicenseConfiguration.html
* https://docs.aws.amazon.com/license-manager/latest/APIReference/API_UpdateLicenseConfiguration.html
* https://docs.aws.amazon.com/license-manager/latest/APIReference/API_DeleteLicenseConfiguration.html

**Comment by gazoakley** [comment]
I'm planning to work on this.

**Comment by bflad** [comment]
The `aws_licensemanager_license_configuration` resource has been released in [version 1.53.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1530-december-20-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by bflad** [comment]
This has been released in [version 1.54.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1540-december-21-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
